android/src/main/java/com/facebook/reactnative/androidsdk/FBSettingsModule.java

```diff
@@ -30,24 +33,17 @@ public String getName() {
         return NAME;
     }
 
-    /**
-     * Sets data processing options
-     * @param options list of the options
-     */
-    @ReactMethod
-    public void setDataProcessingOptions(@Nullable String[] options) {
-        FacebookSdk.setDataProcessingOptions(options, 0, 0);
-    }
-
     /**
      * Sets data processing options with country and state
      * @param options list of the options
      * @param country code of the country
      * @param state code of the state
      */
     @ReactMethod
-    public static void setDataProcessingOptionsExtra(@Nullable String[] options, int country, int state) {
-        FacebookSdk.setDataProcessingOptions(options, country, state);
+    public void setDataProcessingOptions(ReadableArray options, int country, int state) {
+        List<String> opts = Utility.reactArrayToStringList(options);
+        String[] optsArray = opts.toArray(new String[0]);
```

Comment:
I thought this construct was odd then when I went to check it, it appears your stackoverflow skills are the same as mine - https://stackoverflow.com/a/4042464/9910298 - or you just do this often enough you already knew the optimal conversion idiom ;-)

Reply:
Java is not my strongest language so I also find it odd, but When in Rome... :P